src/autopr/SubModsUpdate.d

```diff
@@ -0,0 +1,214 @@
+        string tree_sha   = this.tree_sha;
+        string pr_msg_part;
+
+        if (this.updates.length == 0)
+            return;
+
+        foreach (update; this.updates)
+            this.addUpdateCommit(con, fork_owner, fork_name, update, tree_sha, commit_sha, pr_msg_part);
+
+        import vibe.data.json;
+
+        Json reference;
+
+        // If you change this branch name, also change it in github.d
+        auto refname = "refs/heads/neptune-update";
+
+        // Always try updating first. Chances are that the branch already exists
+        try
+        {
+            writefln("%s> Updating reference %s", this.repo, refname);
+            reference = con.updateReference(fork_owner,
+                fork_name, refname, commit_sha);
+        }
+        catch (HTTPAPIException exc)
+        {
+            writef("%s> Failed to update reference, trying create ... ",
+                this.repo);
+
+            stdout.flush();
+
+            reference = con.createReference(fork_owner,
+                fork_name, refname, commit_sha);
+
+            writefln("Ok");
+        }
+
+        auto content = format(
+"This PR has been automatically created by *neptune-autopr* and
+updates the following submodules:
+
+%s
+You can modify this behavior using your `.neptune.yml` file
+as described in the [documentation](https://github.com/sociomantic-tsunami/neptune/blob/master/doc/neptune-metadata.rst#automatic-update-prs).",
+        pr_msg_part);
+
+        if (this.pr_number.isNull)
+        {
+            writefln("%s> Creating PR ...", this.repo);
+            auto pr = con.createPullrequest(this.repo.owner, this.repo.name,
+                PRTitle, content, fork_owner,  this.branch, refname);
+            this.pr_number = pr["number"].get!int;
+        }
+        else
+        {
+            writefln("%s> Adding comment to PR %s", this.repo, this.pr_number);
+            auto pr = con.updatePullrequest(this.repo.owner, this.repo.name,
+                this.pr_number, PRTitle, content);
+
+            auto msg = format("This PR has been updated: \n\n%s",
+                pr_msg_part);
+
+            con.addComment(pr["id"].get!int, msg);
+        }
+    }
+
+    void addUpdateCommit ( ref HTTPConnection con, string fork_owner,
+        string fork_name, SubModUpdate update, ref string tree_sha,
+        ref string commit_sha, ref string pr_msg_part )
+    {
+        import autopr.github;
+        import std.stdio;
+        import std.format;
+
+        // Create Tree with updated submodule
+        auto tree = con.createTree(fork_owner,
+            fork_name, tree_sha,
+            format("submodules/%s", update.repo.name),
+            update.target.sha, Type.Submodule);
+
+        tree_sha = tree["sha"].get!string;
+
+        writefln("%s> Created updated tree for %s (%s)",
+            this.repo, update.repo.name, tree_sha);
+
+        auto commit_msg = format("Advance %s from %s to %s\n\n%s %s(%s)...%s(%s)",
```

Comment:
VERY bikesheddy, but I'd rather have `Update` than `Advance`.  But that's just because that matches the style that some of my projects are already using ;-)

Reply:
you'll need to fight with @ben-palmer-sociomantic over this one

Reply:
Semantically, I think `Update` is unarguable fact, whereas `Advance` is a matter of opinion ;-)  But it's VERY bikesheddy, so feel free to ignore.